app/assets/javascripts/discourse/app/components/edit-category-general.gjs

```diff
@@ -0,0 +1,347 @@
+
+    // if editing a category, don't include its color:
+    return categories
+      .map((c) => {
+        return categoryId &&
+          categoryColor.toUpperCase() === c.color.toUpperCase()
+          ? null
+          : c.color.toUpperCase();
+      })
+      .compact();
+  }
+
+  @cached
+  get parentCategories() {
+    return this.site
+      .get("categoriesList")
+      .filter((c) => c.level + 1 < this.siteSettings.max_category_nesting);
+  }
+
+  @action
+  categoryBadgePreview(transientData) {
+    const category = this.args.category;
+
+    const previewCategory = Category.create({
+      name: transientData.name || i18n("category.untitled"),
+      color: transientData.color,
+      id: category.id,
+      text_color: category.text_color,
+      parent_category_id: parseInt(category.get("parent_category_id"), 10),
+      read_restricted: category.get("read_restricted"),
+    });
+
+    return categoryBadgeHTML(previewCategory, {
+      link: false,
+      previewColor: true,
+      styleType: transientData.style_type,
+      styleEmoji: transientData.style_emoji,
+      styleIcon: transientData.style_icon,
+    });
+  }
+
+  // We can change the parent if there are no children
+  @cached
+  get subCategories() {
+    if (this.args.category.isNew) {
+      return null;
+    }
+    return Category.list().filter(
+      (category) => category.get("parent_category_id") === this.args.category.id
+    );
+  }
+
+  @cached
+  get showDescription() {
+    const category = this.args.category;
+    return (
+      !category.isUncategorizedCategory && category.id && category.topic_url
+    );
+  }
+
+  @action
+  showCategoryTopic() {
+    window.open(this.args.category.get("topic_url"), "_blank").focus();
+    return false;
+  }
+
+  @action
+  updateField(name, value, { set }) {
+    set(name, value);
+    this[name] = value;
+  }
+
+  @action
+  updateColor(field, newColor) {
+    field.set(newColor.replace("#", ""));
+  }
+
+  get categoryDescription() {
+    if (this.args.category.description) {
+      return htmlSafe(this.args.category.description);
+    }
+
+    return i18n("category.no_description");
+  }
+
+  get canSelectParentCategory() {
+    return !this.args.category.isUncategorizedCategory;
+  }
+
+  get panelClass() {
+    const isActive = this.args.selectedTab === "general" ? "active" : "";
+    return `edit-category-tab edit-category-tab-general ${isActive}`;
+  }
+
+  <template>
+    <div class={{this.panelClass}}>
+      {{#if this.showWarning}}
+        <@form.Alert @type="warning" @icon="triangle-exclamation">
+          {{htmlSafe
+            (i18n
+              "category.uncategorized_general_warning"
+              settingLink=this.uncategorizedSiteSettingLink
+              customizeLink=this.customizeTextContentLink
+            )
+          }}
+        </@form.Alert>
+      {{/if}}
+
+      <PluginOutlet
+        @name="category-name-fields-details"
+        @outletArgs={{hash category=@category}}
```

Comment:
we should expose form object in the outlet so we can start building using form-kit